src/subdomains/generic/user/models/user-data/user-data.controller.ts

```diff
@@ -35,6 +39,31 @@ export class UserDataController {
     return this.userDataRepo.find();
   }
 
+  // --- API KEYS --- //
+  @Post('apiKey/CT')
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.USER))
+  @ApiCreatedResponse({ type: ApiKeyDto })
+  async createApiKey(@GetJwt() jwt: JwtPayload, @Query() filter: HistoryFilter): Promise<ApiKeyDto> {
+    return this.userDataService.createApiKey(jwt.user, filter);
+  }
+
+  @Delete('apiKey/CT')
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.USER))
+  @ApiOkResponse()
+  async deleteApiKey(@GetJwt() jwt: JwtPayload): Promise<void> {
+    return this.userDataService.deleteApiKey(jwt.user);
+  }
+
+  @Put('apiFilter/CT')
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.USER))
+  @ApiOkResponse({ type: String, isArray: true })
+  async updateApiFilter(@GetJwt() jwt: JwtPayload, @Query() filter: HistoryFilter): Promise<HistoryFilterKey[]> {
+    return this.userDataService.updateApiFilter(jwt.user, filter);
+  }
```

Comment:
I would keep it in the UserController (we don't have anything public in the UserDataController), the user doesn't know about UserData

Reply:
`UserRole.ACCOUNT`?

Reply:
Ah ok good point so you would then say, we change it instantly in userController, from userService to userDataService methods?

Reply:
Or should we give the option to give the version number as parameter? 🤔

Reply:
I think I would create the API key on user data, for all new keys. For existing ones, we would need to update/delete it on the user. WDYT?

Reply:
Jap would say the same 👍